x-pack/plugins/elastic_assistant/server/routes/attack_discovery/post_attack_discovery.ts

```diff
@@ -134,23 +151,120 @@ export const postAttackDiscoveryRoute = (
+              throw new Error(
+                `Could not update attack discovery for connectorId: ${apiConfig.connectorId}`
+              );
+            } else {
+              currentAd = ad;
+            }
           }
 
-          const { alertsContextCount, attackDiscoveries } = JSON.parse(rawAttackDiscoveries);
+          toolInstance
+            ?.invoke('')
+            .then(async (rawAttackDiscoveries) => {
+              const getDataFromJSON = () => {
+                const { alertsContextCount, attackDiscoveries } = JSON.parse(rawAttackDiscoveries);
+                return { alertsContextCount, attackDiscoveries };
+              };
+
+              const endTime = moment();
+              const durationMs = endTime.diff(startTime);
+
+              if (rawAttackDiscoveries == null) {
+                throw new Error('tool returned no attack discoveries');
+              }
+              const updateProps = {
+                ...getDataFromJSON(),
+                status: attackDiscoveryStatus.succeeded,
+                generationIntervals: addGenerationInterval(currentAd.generationIntervals, {
+                  durationMs,
+                  date: new Date().toISOString(),
+                }),
+                id: attackDiscoveryId,
+                replacements: latestReplacements,
+                backingIndex: currentAd.backingIndex,
+              };
+
+              await dataClient?.updateAttackDiscovery({
+                attackDiscoveryUpdateProps: updateProps,
+                authenticatedUser,
+              });
+
+              telemetry.reportEvent(ATTACK_DISCOVERY_SUCCESS_EVENT.eventType, {
+                actionTypeId: apiConfig.actionTypeId,
+                alertsContextCount: updateProps.alertsContextCount,
+                alertsCount: uniq(
+                  updateProps.attackDiscoveries.flatMap(
+                    (attackDiscovery: AttackDiscovery) => attackDiscovery.alertIds
+                  )
+                ).length,
+                configuredAlertsCount: size,
+                discoveriesGenerated: updateProps.attackDiscoveries.length,
+                durationMs,
+                model: apiConfig.model,
+                provider: apiConfig.provider,
+              });
+            })
+            .catch(async (err) => {
+              logger.error(err);
+              const error = transformError(err);
+              telemetry.reportEvent(ATTACK_DISCOVERY_ERROR_EVENT.eventType, {
+                actionTypeId: apiConfig.actionTypeId,
+                errorMessage: error.message,
+                model: apiConfig.model,
+                provider: apiConfig.provider,
+              });
+              await dataClient?.updateAttackDiscovery({
+                attackDiscoveryUpdateProps: {
+                  attackDiscoveries: [],
+                  status: attackDiscoveryStatus.failed,
+                  id: attackDiscoveryId,
+                  replacements: latestReplacements,
```

Comment:
Consider persisting empty `replacements`, because `attackDiscoveries` is empty, so the replacements won't be used

Reply:
ok i removed it from the create call entirely. thanks https://github.com/elastic/kibana/pull/184949/commits/0ac25b987bddd781fc5f7f9212a721bbe3ea10dd